### PR TITLE
fix some level 1 warnings

### PR DESCRIPTION
**Pull request by bkentel**
fix a bunch of warnings related to C style comments and using floats in boolean expressions.

### CODE REVIEW

File: src/trapfunc.cpp [hunk 1]
```diff
@@ -822,17 +822,23 @@ void trapfunc::lava(Creature *c, int x, int y)
...
 
-void trapfunc::sinkhole(Creature *c, int /*x*/, int /*y*/)
-{
-    if (c != &g->u) {
+void trapfunc::sinkhole(
+    Creature * //c
+  , int //x
+  , int //y
+) {
+    //if (c != &g->u) {
```
Comment (kevingranade) [inline src/trapfunc.cpp]: generally speaking, never comment out code.  If it shouldn't be there, delete it.

File: src/trapfunc.cpp [hunk 2]
```diff
@@ -822,17 +822,23 @@ void trapfunc::lava(Creature *c, int x, int y)
 }
 
 // STUB
-void trapfunc::portal(Creature */*c*/, int /*x*/, int /*y*/)
-{
+void trapfunc::portal(
```
Comment (kevingranade) [inline src/trapfunc.cpp]: You're throwing the coding style out the window here.  Plus I'm not sure what warnings this would be silencing.

**Comment by bkentel** [comment]
*/*c makes it complain about embedded /* and / or */
how would you suggest dropping unused params in general? just add an extra space here and keep the /* */ ?

**Comment by KA101** [comment]
>unused params

Make a use for 'em, or call 'em void from the start of the function.

**Comment by bkentel** [comment]
even when it's not my code? I should rip out empty bits like this when e.g. fixing warnings? I have no problem with that if that's the policy.

**Comment by KA101** [comment]
If you're working with a PR, compile it, and it errors (warnings == errors for buildbot purposes), you get to fix that error.  (Or abandon the PR, granted, that works too, but may not be what you want.  Sometimes, someone else can help with your PR, and sometimes not.)

We run -werror on the buildbot so if Jenkins/clang sees something it doesn't like, such as an unused param, FAIL goes the build.

**Comment by bkentel** [comment]
In this case it wasn't the unused param (there isn't one stricktly speaking) per se, but the specific syntax that *MSVC* warned about.

In this particular case, what would you say should be done? I can just insert a space to fix the warning I think.

**Comment by KA101** [comment]
I don't know, I don't use MSVC.

**Comment by bkentel** [comment]
Should be fixed now without disrupting style.

File: src/vehicle.cpp [hunk 3]
```diff
@@ -5393,7 +5393,7 @@ void vehicle::open_or_close(int part_index, bool opening)
       int ydiff = parts[next_index].mount_dy - parts[part_index].mount_dy;
       if((xdiff * xdiff + ydiff * ydiff == 1) && // (x^2 + y^2) == 1
               (part_info(next_index).id == part_info(part_index).id) &&
-              (parts[next_index].open == opening ? 0 : 1)) {
+              (parts[next_index].open == 1 && opening)) {
```
Comment (BevapDin) [inline src/vehicle.cpp]: If `opening` is false, the condition will not be true ever. This is not the same behavior as the current code. The current code checks a *relation* between `.open` and `opening`. The changed code does not check that relation, it check their state completely separated.

Comment (bkentel) [inline src/vehicle.cpp]:
```
parts[next_index].open == opening ? 0 : 1

int a = parts[next_index].open
bool b = opening

(a == b) ? 0 : 1 <==> (a && b) ? 0 : 1

parts[next_index].open == 1 && opening

(a == 1) && b <==> (a && b)
```

So, yes, sorry the condition is backwards; it should be

```cpp
!(parts[next_index].open  && opening)
```
Good catch; was a careless change on my part.

Comment (BevapDin) [inline src/vehicle.cpp]: > (a == b) ? 0 : 1 <==> (a && b) ? 0 : 1

Ähm, No. `false == false` is true, but `false && false` is false.

How about `.open != opening` without the ternary?

```
     (parts[next_index].open == opening ? 0 : 1)
<==> (parts[next_index].open != opening ? 1 : 0)
<==> (parts[next_index].open != opening ? true : false)
<==> (parts[next_index].open != opening)
```
The only thing is `.open` is an int, not a bool. That might trigger the warning you were about to remove. In that case you might want to warp it: `(.open == 1) != opening`

Comment (bkentel) [inline src/vehicle.cpp]: so then xor it is

**Pull request merged**